commit b4d47fc54edb816288d41af5510d02e34f8d16db (refs/changes/87/3187/4)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-11T19:03:58+01:00 (1 year, 9 months ago)
    
    TOOLS-1972 'sdcadm avail' crashes when there are not available updates
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/cli/do_avail.js b/lib/cli/do_avail.js
index cf6a528..6c8b476 100644
--- a/lib/cli/do_avail.js
+++ b/lib/cli/do_avail.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -102,7 +102,8 @@ Available.prototype.execute = function cExecute(opts, args, cb) {
         }
     ]}, function availCb(err) {
         if (err) {
-            return cb(err);
+            cb(err);
+            return;
         }
 
         var rows = [];
@@ -169,25 +170,22 @@ Available.prototype.execute = function cExecute(opts, args, cb) {
                     process.stdout.write(JSON.stringify(k) + '\n');
                 });
             }
-            return cb();
+            cb();
+            return;
         }
 
-        var validFieldsMap = {};
-
-        rows.forEach(function (v) {
-            var k;
-            for (k in v) {
-                validFieldsMap[k] = true;
-            }
-        });
+        if (!rows.length) {
+            console.log('Up-to-date.');
+            cb();
+            return;
+        }
 
         tabula(rows, {
             skipHeader: opts.H,
             columns: columns,
-            sort: sort,
-            validFields: Object.keys(validFieldsMap)
+            sort: sort
         });
-        return cb();
+        cb();
     });
 };
 
